### PR TITLE
elf-util: drop assertion for metadata in report_module_metadata()

### PR DESCRIPTION
**Pull request by yuwata**
Fixes RHBZ#2134741 (https://bugzilla.redhat.com/show_bug.cgi?id=2134741).

### CODE REVIEW

**Review by bluca — APPROVED** [review]

File: src/shared/elf-util.c [hunk 1]
```diff
@@ -293,11 +293,13 @@ static char* build_package_reference(
 static void report_module_metadata(StackContext *c, const char *name, JsonVariant *metadata) {
         assert(c);
         assert(name);
-        assert(metadata);
 
         if (!c->f)
                 return;
 
+        if (!metadata)
+                return;
```
Comment (poettering) [inline src/shared/elf-util.c]: is this the right approach? if we have neither buildid nor the packaging info, should we hide the module name or show it? with this change you hide it altogether?

@keszybz what is the intended behaviour here?

Comment (keszybz) [reply src/shared/elf-util.c]: I think we should still print the module name. It is useful, even if only to figure out what modules are without appropriate metadata.

Comment (yuwata) [reply src/shared/elf-util.c]: Now the module name is printed even if the JSON variant is NULL. PTAL.

**Comment by poettering** [comment]
lgtm

**Comment by keszybz** [comment]
I don't think our CI cares about the output format at all. If it compiles, it's good to go.

**Pull request merged**